homeassistant/components/freedompro/config_flow.py

```diff
@@ -0,0 +1,77 @@
+"""Config flow to configure Freedompro."""
+from pyfreedompro import get_list
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.helpers import aiohttp_client
+
+from .const import DOMAIN  # pylint:disable=unused-import
+from .const import FREEDOMPRO_URL
+
+STEP_USER_DATA_SCHEMA = vol.Schema({"api_key": str})
+
+
+class Hub:
+    """Freedompro Hub class."""
+
+    def __init__(self, hass, api_key):
+        """Freedompro Hub class init."""
+        self._hass = hass
+        self._api_key = api_key
+        self._url = FREEDOMPRO_URL
+
+    async def authenticate(self) -> bool:
+        """Freedompro Hub class authenticate."""
+        result = await get_list(
+            aiohttp_client.async_get_clientsession(self._hass), self._api_key
+        )
+        return result
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate api key."""
+    hub = Hub(hass, data["api_key"])
+    result = await hub.authenticate()
+    if result["state"] is False:
+        if result["code"] == -201:
+            raise InvalidAuth
+        if result["code"] == -200:
+            raise CannotConnect
+    return {"title": "Freedompro"}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_UNKNOWN
```

Comment:
Should be the same as in [the documentation](https://github.com/home-assistant/home-assistant.io/pull/16492/files#diff-d4aee1a7bdb0843d4cc9f6ff966319ac817968e14c71a7e940463f6ffe3679e3R7).

```suggestion
    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
```